docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,698 @@ id: pillars-turbomodules
+
+##### RTNCalculator.h
+
+```objc title="RTNCalculator.h"
+#import <React/RCTBridgeModule.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RTNCalculator : NSObject <RCTBridgeModule>
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+This file defines the interface for the `RTNCalculator` module. Here, we can add any native method we may want to invoke on the view. For this guide, we don't need anything, therefore the interface is empty.
+
+
+##### RTNCalculator.mm
+
+```objc title="RTNCalculator.mm"
+#import "RTNCalculatorSpec.h"
+#import "RTNCalculator.h"
+
+@implementation RTNCalculator
+
+RCT_EXPORT_MODULE(RTNCalculator)
+
+RCT_REMAP_METHOD(add, addA:(NSInteger)a
+                      andB:(NSInteger)b
+                withResolver:(RCTPromiseResolveBlock) resolve
+                withRejecter:(RCTPromiseRejectBlock) reject)
+{
+    NSNumber *result = [[NSNumber alloc] initWithInteger:a+b];
+    resolve(result);
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+    return std::make_shared<facebook::react::NativeCalculatorSpecJSI>(params);
+}
+
+@end
+```
+
+The most important call is to the `RCT_EXPORT_MODULE`, which is required to export the module so that React Native can load the TurboModule.
+
+Then the `RCT_REMAP_METHOD` macro is used to expose the `add` method.
+
+:::info
+There are other macros that can be used to export modules and methods. You view the code that specifies them [here](https://github.com/facebook/react-native/blob/main/React/Base/RCTBridgeModule.h).
+:::
+
+TODO: more description?
+
+### Android
+
+Android follows similar steps to iOS. We have to generate the code for Android, and then we have to write some native code to make it work.
+
+#### Generate the Code - Android
+
+To generate the code for Android, we need to manually invoke CodeGen. This is done similarly to what we did for iOS: first, we need to add the package to the app and then we need to invoke a script.
+
+```sh title="Running CodeGen for Android"
+cd MyApp
+yarn add ../RTNCalculator
+cd android
+./gradlew generateCodegenArtifactsFromSchema --rerun-tasks
+```
+
+This script first adds the package to the app, in the same way iOS does. Then, after moving to the `android` folder, it invokes a Gradle task to create the generated code.
+
+:::note
+To run **CodeGen**, you need to enable the **New Architecture** in the Android app. This can be done by opening the `gradle.properties` files and by switching the `newArchEnabled` property from `false` to `true`.
+:::
+
+The generated code is stored in the `MyApp/node_modules/rtn-calculator/android/build/generated/source/codegen` folder and it has this structure:
+
+TODO: ordering issue. I can't actually get codegen to work at this point. Only works once I add the native files following. Probably similar issue for Fabric guide. Including generated file structure for later ease of use:
```

Comment:
Do you have errors here or the CodeGen just does not generate anything?

Reply:
We went further in the exploration and you were right. Android requires other files in order to be picked up by the Codegen. Specifically, it needs also:
-  The Android Manifest
- A file that implements the `ReactPackage` interface or that extends the `TurboReactPackage` class

Reply:
From the conversation in Discord, I'm not sure still exactly how much code I would write for that `TurboReactPackage` class before running Codegen; especially since I know I've run into issues on the Android side with creating an incomplete or incorrect interface before and then changing it.